augur/db/defaults.py

```diff
@@ -0,0 +1,7 @@
+# Table names
+DEFAULT_IMPORTED_METADATA_TABLE = 'metadata'
+DEFAULT_IMPORTED_SEQUENCES_TABLE = 'sequences'
```

Comment:
+1 to unifying in a single table. This how GenBank records, GISAID NJSON records, etc. already work and, like @joverlee521 mentions, how our curate logic thinks about metadata and sequences, where a sequence is another attribute of a strain record. The choice to store metadata as TSV and sequences as FASTA was based on the desire to keep intermediate and final outputs from Augur in standard bioinformatics formats so other tools could parse them. Inside the database, that split doesn't make as much sense.

How would this single table implementation relate to prior work with SFS's sequence database?

Reply:
> How would this single table implementation relate to prior work with SFS's sequence database?

This would be much simpler than the [SFS ID3C schema](https://github.com/seattleflu/documentation/blob/c0b57e54d10bb32e795ccc9f5d73616bc53ce779/id3c-warehouse-schema.pdf) since ID3C had to support one-to-many relationships. I think we only expect one-to-one between metadata and sequences here?

Reply:
I would expect one-to-many relationships if we expect `augur db` to work with multisegmented viruses:

* e.g. One influenza A isolate links out to PB2,PB1,PA,HA,NA,NP,NS sequences.

However, if we're using Accession IDs (and not strain names) then ignore my comment above. There should be a one-to-one match of Accession-sequence. 

I would expect many to one in the case of:

* Multiple isolates from the same barn/hospital/home have the same consensus sequence.
* Convergent evolution

Reply:
> I think we only expect one-to-one between metadata and sequences here?

Setting the expectation here is key. I'm not sure what to expect as I don't have much experience using these file formats in practice.

If we intend to support situations where there are duplicates or mismatches, even for the curate step (e.g. metadata de-duplication by taking the most recent metadata entry), the behavior should be customizable by the user.

We can easily store raw data as two separate tables and create a [view](https://en.wikipedia.org/wiki/View_(SQL)) that combines both for easy query-ing. It can be customized based on how duplicates or mismatches should be handled.

---

@joverlee521 separately, we were discussing that maybe even `augur curate` could use the database internally. Things like column indexing could give a performance boost.